tests/wpunit/UserObjectQueriesTest.php

```diff
@@ -1553,4 +1553,154 @@ public function testGetUserByIdTypesAsSubscriber() {
 
 	}
 
+	public function testQueryUsersAsPublicUserShouldReturnOnlyPublishedAuthors() {
+
+		$alphabet = range( 'A', 'Z' );
+		$published_users = [];
+		$unpublished_users = [];
+		foreach ( $alphabet as $letter ) {
+			$unpublished_users[] = $this->factory()->user->create([
+				'user_login' => 'unpublished_' . $letter,
+				'user_email' => $letter . '_unpublishded@example.com'
+			]);
+			$author_id = $this->factory()->user->create([
+				'user_login' => 'published_' . $letter,
+				'user_email' => $letter . '_published@example.com',
+				'role' => 'administrator'
+			]);
+
+			$published_users[] = $author_id;
+			$this->factory()->post->create([
+				'post_status' => 'publish',
+				'post_title' => $letter . '_Post',
+				'post_author' => $author_id,
+				'role' => 'administrator'
+			]);
+		}
+
+		$query = '
```

Comment:
Why query 2 or 3 different ways? You are only checking the ids match the correct type of author/publisher.

Reply:
Ah yes. I should explain more in the PR description. 

Let's say we have the following list of users, conveniently named with one letter of the alphabet: 

```json
A published
B published
C published
D published
E published
F published
G published
H published
I published
J published
K published
L not-published
M not-published
N not-published
O not-published
P not-published
Q not-published
R not-published
S not-published
T not-published
U not-published
V not-published
W not-published
X not-published
Y not-published
Z not-published
```

The bug right now, is that SQL queries the number of items that were asked for (+1 for cursor pagination), then the Model Layer determines if the object can be returned. 

So, asking for `first:10` will return authors `A-K`, slice `K` off the list (and set `hasNextPage` as true) and the Model Layer will see that the authors are all published and can be returned so we will get authors `A-J` returned. 

Now, if we ask for `last:10` the bug can be seen. 

SQL will ask for authors `P-Z`, slice off `P` (and set `hasPreviousPage` as true) then pass the rest to the Model Layer. Since none of them are published authors, the Model Layer will prevent them from being returned to a non-authenticated user and it will appear as if there are no users at all. 

What should happen, though, is that a public request for `last: 10` should return the `last 10 users that the requesting user has permission to see`. 

So the expectation is that the public user should get users `B-K` returned when asking for `{ users(last:10) { ... } }`